cpp/src/arrow/dataset/discovery.h

```diff
@@ -227,16 +227,6 @@ class ARROW_DS_EXPORT FileSystemDatasetFactory : public DatasetFactory {
       std::shared_ptr<fs::FileSystem> filesystem, fs::FileSelector selector,
       std::shared_ptr<FileFormat> format, FileSystemFactoryOptions options);
 
-  /// \brief Build a FileSystemDatasetFactory from an uri including filesystem
-  /// information.
-  ///
-  /// \param[in] uri passed to FileSystemDataset
-  /// \param[in] format passed to FileSystemDataset
-  /// \param[in] options see FileSystemFactoryOptions for more information.
-  static Result<std::shared_ptr<DatasetFactory>> Make(std::string uri,
```

Comment:
Removed this method from the PR since some of the factory options might vary according to URI, e.g. We may want to enable partitioning when reading folders and disable partitioning when reading a single file. But before URI is resolved we don't even know it's a file or a folder. We may add URI based API back once we have a better design on it.